controller/api/public/stat_summary.go

```diff
@@ -292,19 +364,21 @@ func buildRequestLabels(req *pb.StatSummaryRequest) (labels model.LabelSet, labe
 
```

Comment:
I think we also need to make a special exception when the `ToResource` is an authority, since we won't have any `dst_authority` labels, and we won't have `dst_ns` labels when the authority is external to the k8s cluster. So I recommend changing the `ToResource` case to:

```
	case *pb.StatSummaryRequest_ToResource:
		labelNames = promLabelNames(req.Selector.Resource)

		if req.GetSelector().GetResource().GetType() != k8s.Authority {
			labels = labels.Merge(promDstLabels(out.ToResource))
		}
		labels = labels.Merge(promLabels(req.Selector.Resource))
		labels = labels.Merge(promDirectionLabels("outbound"))
```

Reply:
I've refactored some of the label methods, maybe this will help. They are kind of complicated.

I think we can defer the setting of the labels to `promDstLabels`, since if we're querying a `ToAuthority`, we'd still need to set the label `{authority=to_authority_name}`, which won't be set by `req.Selector.Resource.Name` (we need to get it from `out.ToResource.Name`)

Currently promDstLabels does take care of checking to make sure we don't add a `dst_authority` or a `dst_ns` in the case of a `ToAuthority` query. And it sets the `{authority=to_authority}` that we want to filter by:

```
// query a named resource
func promDstLabels(resource *pb.Resource) model.LabelSet {
	set := model.LabelSet{}
	if resource.Name != "" {
		if resource.Type == k8s.Authority {
			set[promResourceType(resource)] = model.LabelValue(resource.Name)
		} else {
			set["dst_"+promResourceType(resource)] = model.LabelValue(resource.Name)
			if shouldAddNamespaceLabel(resource) {
				set[dstNamespaceLabel] = model.LabelValue(resource.Namespace)
			}
		}
	}

	return set
}
```